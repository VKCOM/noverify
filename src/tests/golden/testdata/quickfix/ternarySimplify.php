<?php

$a = 5;
function f(): int {
    global $a;
    $a++;
    return $a;
}

function ternarySimplify() {
    $x = 0;
    $y = 0;

    $_ = $x ? $x : 5;
    // ok, f not pure
    $_ = f() ? f() : 5;

    $_ = isset($x) ? $x : $y;
}
