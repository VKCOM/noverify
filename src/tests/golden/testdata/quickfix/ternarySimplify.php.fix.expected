<?php

$a = 5;
function f() {
    $a++;
    return $a;
}

function ternarySimplify() {
    $z = $x ?: 5;
    // f not pure
    $z = f() ? f() : 5;

    $z = $x ?? $y;
}
